pkg/cluster/majorversionupgrade.go

```diff
@@ -67,12 +123,23 @@ func (c *Cluster) majorVersionUpgrade() error {
 	}
 
 	desiredVersion := c.GetDesiredMajorVersionAsInt()
+	isUpgradeSuccess := true
+	lastMajorUpgradeAnnotations := c.getLastMajorUpgradeAnnotations()
 
 	if c.currentMajorVersion >= desiredVersion {
+		if lastMajorUpgradeAnnotations["last-major-upgrade-success"] == "" {
+			c.annotatePostgresResource(isUpgradeSuccess)
+			c.logger.Info("update last major upgrade success annotation to current timestamp as it was not set")
+		}
 		c.logger.Infof("cluster version up to date. current: %d, min desired: %d", c.currentMajorVersion, desiredVersion)
 		return nil
 	}
 
+	if lastMajorUpgradeAnnotations["last-major-upgrade-failure"] != "" {
```

Comment:
following up my comment above, if the failure annotations stays in the manifest, we have to compare the time with the last success. If failure is older than success major version we can proceed and should not exit here.